private/bufpkg/bufpolicy/errors.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020-2025 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufpolicy
+
+import (
+	"strings"
+
+	"github.com/bufbuild/buf/private/bufpkg/bufparse"
+	"github.com/bufbuild/buf/private/pkg/uuidutil"
+	"github.com/google/uuid"
+)
+
+// DigestMismatchError is the error returned if the Digest of a downloaded Policy
+// does not match the expected digest in a buf.lock file.
+type DigestMismatchError struct {
+	FullName       bufparse.FullName
+	CommitID       uuid.UUID
+	ExpectedDigest Digest
+	ActualDigest   Digest
+}
```

Comment:
I don't think this is something that should happen in this PR, but thoughts on moving `DigestMismatchError` into a shared package under `bufpkg`, similar to `bufparse`?